demo/src/BaseAgent.ts

```diff
@@ -10,46 +36,165 @@ const bcovrin = `{"reqSignature":{},"txn":{"data":{"data":{"alias":"Node1","blsk
 
+const indyNetworkConfig = {
+  // Need unique network id as we will have multiple agent processes in the agent
+  id: randomUUID(),
+  genesisTransactions: bcovrin,
+  indyNamespace: 'bcovrin:test',
+  isProduction: false,
+  connectOnStartup: true,
+} satisfies IndySdkPoolConfig | IndyVdrPoolConfig
+
+type DemoAgent = Agent<ReturnType<typeof getIndySdkModules> | ReturnType<typeof getSharedComponentModules>>
+
 export class BaseAgent {
   public port: number
   public name: string
   public config: InitConfig
-  public agent: Agent
+  public agent: DemoAgent
+  public anonCredsIssuerId: string
+  public usesSharedComponents: boolean
 
-  public constructor(port: number, name: string) {
+  public constructor({
+    port,
+    name,
+    useSharedComponents,
+  }: {
+    port: number
+    name: string
+    useSharedComponents: boolean
+  }) {
     this.name = name
     this.port = port
 
-    const config: InitConfig = {
+    const config = {
       label: name,
       walletConfig: {
         id: name,
         key: name,
       },
-      publicDidSeed: '6b8b882e2618fa5d45ee7229ca880083',
-      indyLedgers: [
-        {
-          genesisTransactions: bcovrin,
-          id: 'greenlights' + name,
-          indyNamespace: 'greenlights' + name,
-          isProduction: false,
-        },
-      ],
+      publicDidSeed: 'afjdemoverysercure00000000000000',
       endpoints: [`http://localhost:${this.port}`],
       autoAcceptConnections: true,
-      autoAcceptCredentials: AutoAcceptCredential.ContentApproved,
-      autoAcceptProofs: AutoAcceptProof.ContentApproved,
-    }
+    } satisfies InitConfig
 
     this.config = config
 
-    this.agent = new Agent({ config, dependencies: agentDependencies })
+    // TODO: do not hardcode this
+    this.anonCredsIssuerId = '2jEvRuKmfBJTRa7QowDpNN'
+    this.usesSharedComponents = useSharedComponents
+
+    this.agent = new Agent({
+      config,
+      dependencies: agentDependencies,
+      modules: useSharedComponents ? getSharedComponentModules() : getIndySdkModules(),
+    })
     this.agent.registerInboundTransport(new HttpInboundTransport({ port }))
     this.agent.registerOutboundTransport(new HttpOutboundTransport())
   }
 
   public async initializeAgent() {
     await this.agent.initialize()
+
+    // FIXME:
+    // We need to make sure the key to submit transactions is created. We should update this to use the dids module, and allow
+    // to add an existing did based on a seed/secretKey, and not register it on the the ledger. However for Indy SDK we currently
+    // use the deprecated publicDidSeed property (which will register the did in the wallet), and for Askar we manually create the key
+    // in the wallet. Additional issue is that when creating a key in askar using the seed, it will give different results than indy-sdk
+    // but if we create the key in askar from the seed, but use it as the secret key, it will work.
```

Comment:
I think we can attempt to review/merge #1301 before this one, so we can have it clean and be able to do the demo right from the main branch.

Reply:
Agreed! Just approved